### ISSUE TITLE
 The request to rTorrent has timed out.

### ISSUE THREAD

**Issue by terrorindeed** [Bercik1337/rt-auto-install#35]
When adding Torrents, labelling exisiting torrents and a few other common tasks, I get this error. 
When looking at the log, this error pops up every 20 minutes on que.

Thoughts?

Ubuntu Server 22.04.1 LTS

**Comment by Bercik1337** [comment]
See `systemctl status rtorrent` if your rt has restarted or something like that. If yes, that's a problem with rtorrent itself.
If it didn't restart it's something between you and rutorrent. Is this local installation, other computer on same network or remote VPS?

**Comment by terrorindeed** [comment]
I checked systemctl , Rtorrent hasn’t restarted. 
This is a local install on dedicated VM. Accessed via local network . Of note, my torrents are from a mounted rclone folder (google drive). 
For example, I cannot navigate to the folder, rutorrent cannot see it. I manually type in /home/tcd/crypt/xxxxxxx/ , force recheck and begin seeding. This vm is a seeding only rtorrent instance.

**Comment by Bercik1337** [comment]
try setting up another VM but with LOCAL storage instead of google drive. we'll see if that's the reason. I could see possibility that latency between the two could cause some problems.

**Comment by terrorindeed** [comment]
I have a few more dedicated VM’s as well with local only storage that suffer the same issue.

**Comment by terrorindeed** [comment]
This is has escalated to the point where rutorrent cannot see rtorrent (it is running / active). 
No connection to rTorrent. Check if it is really running. Check $scgi_port and $scgi_host settings in config.php and scgi_port in rTorrent configuration file.

Both files have been checked and verified. No changes have been made. 
The details on this VM are Ubuntu 22.04.1 LTS, up to date. Php 8.1 
I'm guessing this has something to do with it. Any suggestions?

**Comment by Bercik1337** [comment]
I'll setup VM this weekend. I doubt it's 22.04.1 release fault, other people would report that too. There must be something specific about your system. In touch

**Comment by terrorindeed** [comment]
I cloned an earlier version of one of my VM's. Ubuntu 20.04 LTS, I spun it up earlier this moring and I'm rebuilding it as we speak. Adding 20 torrents at a time. I'm still getting the request to rtorrent has timed out, but the connection seems to re-establish right away. 

Force recheck is working and I'm haven't seen the error 'No connection to rTorrent. Check if it is really running. Check $scgi_port and $scgi_host settings in config.php and scgi_port in rTorrent configuration file' yet.

**Comment by terrorindeed** [comment]
Well, this time around. Things improved 100%, I left the VM alone for 5 minutes, came back and refreshed my webui link. And those 10 UHD discs I added, showed as unlablled and queued. So right now I've got 44.36TB of partially completed or queued torrents. I'm feeling like for my purposes (cloud seeding) something about ubuntu 20.04 LTS works better.

**Comment by terrorindeed** [comment]
Very odd, another error, just randomly. No connection to rtorrent. running sudo systemctl status rtorrent.service yields the following. I also rebooted, to only find the same error.
rtorrent.service - rtorrent (in tmux)
     Loaded: loaded (/etc/systemd/system/rtorrent.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2022-12-04 03:04:38 UTC; 1s ago
    Process: 2255 ExecStart=/usr/bin/tmux -2 new-session -d -s rtorrent rtorrent (code=exited, status=0/SUCCESS)
    Process: 2273 ExecStop=/usr/bin/tmux send-keys -t rtorrent:rtorrent C-q (code=exited, status=1/FAILURE)
   Main PID: 2263 (code=exited, status=0/SUCCESS)

**Comment by terrorindeed** [comment]
rtorrent.service - rtorrent (in tmux)
     Loaded: loaded (/etc/systemd/system/rtorrent.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-12-04 03:10:28 UTC; 1s ago
    Process: 3516 ExecStart=/usr/bin/tmux -2 new-session -d -s rtorrent rtorrent (code=exited, status=0/SUCCESS)
   Main PID: 3526 (tmux: server)
      Tasks: 3 (limit: 37975)
     Memory: 37.8M
     CGroup: /system.slice/rtorrent.service
             ├─3526 /usr/bin/tmux -2 new-session -d -s rtorrent rtorrent
             └─3534 rtorrent

Dec 04 03:10:28 new-rtorrent systemd[1]: Starting rtorrent (in tmux)...
Dec 04 03:10:28 new-rtorrent systemd[1]: Started rtorrent (in tmux).
tcd@new-rtorrent:~$ sudo systemctl status rtorrent.service
● rtorrent.service - rtorrent (in tmux)
     Loaded: loaded (/etc/systemd/system/rtorrent.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Sun 2022-12-04 03:10:29 UTC; 4s ago
    Process: 3516 ExecStart=/usr/bin/tmux -2 new-session -d -s rtorrent rtorrent (code=exited, status=0/SUCCESS)
    Process: 3542 ExecStop=/usr/bin/tmux send-keys -t rtorrent:rtorrent C-q (code=exited, status=1/FAILURE)
   Main PID: 3526 (code=exited, status=0/SUCCESS)

**Comment by terrorindeed** [comment]
Like its now in a boot loop?

**Comment by Bercik1337** [comment]
"1s ago" suggests so.

Please stop rtorrent
`sudo systemctl stop rtorrent.service`
then start it manually without tmux just to see output
`rtorrent`

Sometimes rtorrent can report what's wrong just before it closes itself (like permission problems, no space left etc).

**Comment by terrorindeed** [comment]
indeed. I first checked status, and rtorrent.service had started about 50 minutes ago (no action by myself). I stopped the service and ran rtorrent. Rutorrent is connecting to rtorrent no problem. Could this be an issue with tmux?

**Comment by Bercik1337** [comment]
Sorry I felt asleep. If it's now running, then try starting it `sudo systemctl start rtorrent.service' and do `tmux attach` immediately  after that. If you do this quick enough you should see rtorrent screen, and if it ends abnormally - there should be a message why as well.

As for your initial problem. Few hours ago I've set up ubuntu 20.04, with fresh install of rt-auto-instal. Got few files of Debian, Ubuntu and Mint ISOs and waited 5hours. Not a single problem/timeout/whatsoever.

In other words - I can't reproduce your problem.

**Comment by terrorindeed** [comment]
I attached to the tmux session. But found rtorrent to be often stuck in a reactiving state, I believe. I've moved to setting up rtorrent on my own, in a screen session, for now. And, so far ~3 days on, I've seen request to timeout roughly the same amount of time in rutorrent, but the connection almost immediately reconnects in session. Thanks for the help. I appreciate it.

**Comment by Bercik1337** [comment]
If you ever need to reopen this one, please just ping me.
